packages/dds/tree/src/simple-tree/api/schemaFactory.ts

```diff
@@ -890,3 +896,72 @@ export function markSchemaMostDerived(schema: TreeNodeSchema): void {
 
 	(schema as typeof TreeNodeValid & TreeNodeSchema).markMostDerived();
 }
+
+/**
+ * Binds metadata to a node schema.
+ * @remarks Accomplishes this by creating a new subclass with the specified metadata, strongly typed.
+ *
+ * @param nodeSchema - The node schema to bind metadata to.
+ * @param metadata - The metadata to bind to the node schema.
+ *
+ * @example
+ *
+ * ```typescript
+ * class Point extends withMetadata(
+ * 	schemaFactory.object("point", { x: schemaFactory.number, y: schemaFactory.number }),
+ * 	{
+ * 		description: "A point in 2D space",
+ * 		custom: {
+ * 			... // Your custom metadata properties here
+ * 		},
+ * 	}
+ * ) {}
+ * ```
+ *
+ * @alpha
+ */
+export function withMetadata<
+	TNewCustomMetadata = unknown,
+	TName extends string = string,
+	TKind extends NodeKind = NodeKind,
+	TNode extends TreeNode = TreeNode,
+	TInsertable = never,
+	ImplicitlyConstructable extends boolean = boolean,
+	Info = unknown,
+	TConstructorExtra = never,
+>(
+	nodeSchema: TreeNodeSchemaClass<
+		TName,
+		TKind,
+		TNode,
+		TInsertable,
+		ImplicitlyConstructable,
+		Info,
+		TConstructorExtra
+	>,
+	metadata: NodeSchemaMetadata<TNewCustomMetadata>,
+): TreeNodeSchemaClass<
+	TName,
+	TKind,
+	TNode,
+	TInsertable,
+	ImplicitlyConstructable,
+	Info,
+	TConstructorExtra,
+	TNewCustomMetadata
+> {
+	class Derived extends nodeSchema {
+		public static readonly metadata?: NodeSchemaMetadata<TNewCustomMetadata> | undefined =
```

Comment:
I don't think this should be optional. Its always provided in this case.

Reply:
```suggestion
		public static readonly metadata: NodeSchemaMetadata<TNewCustomMetadata> | undefined =
```